mkosi/__init__.py

```diff
@@ -896,25 +898,49 @@ def prepare_xbootldr(args: CommandLineArguments, loopdev: Optional[str], cached:
         run(["mkfs.fat", "-nXBOOTLDR", "-F32", partition(loopdev, args.xbootldr_partno)])
 
 
-def mkfs_ext4(label: str, mount: str, dev: str) -> None:
-    run(["mkfs.ext4", "-I", "256", "-L", label, "-M", mount, dev])
+def mkfs_ext4(label: str, mount: str, dev: str, args: Optional[List[str]]) -> None:
+    cmdline = ["mkfs.ext4", "-I", "256", "-L", label, "-M", mount]
+    if args:
+        cmdline += args
+    cmdline += [dev]
+    run(cmdline)
 
 
-def mkfs_xfs(label: str, dev: str) -> None:
-    run(["mkfs.xfs", "-n", "ftype=1", "-L", label, dev])
+def mkfs_xfs(label: str, dev: str, args: Optional[List[str]]) -> None:
+    cmdline = ["mkfs.xfs", "-n", "ftype=1", "-L", label]
+    if args:
+        cmdline += args
+    cmdline += [dev]
+    run(cmdline)
 
 
-def mkfs_btrfs(label: str, dev: str) -> None:
-    run(["mkfs.btrfs", "-L", label, "-d", "single", "-m", "single", dev])
+def mkfs_btrfs(label: str, dev: str, args: Optional[List[str]]) -> None:
+    cmdline = ["mkfs.btrfs", "-L", label, "-d", "single", "-m", "single"]
+    if args:
+        cmdline += args
+    cmdline += [dev]
+    run(cmdline)
 
 
 def mkfs_generic(args: CommandLineArguments, label: str, mount: str, dev: str) -> None:
+    mkfs_args = []
+
+    if args.output_format == OutputFormat.gpt_ext4:
+        if (args.distribution in (Distribution.centos, Distribution.centos_epel) and
+            is_older_than_centos8(args.release)):
+            # e2fsprogs in centos7 is too old and doesn't support this feature
+            mkfs_args += ["-O", "^metadata_csum"]
+
+        if args.architecture in ["x86_64", "aarch64"]:
```

Comment:
```suggestion
        if args.architecture in ("x86_64", "aarch64"):
```

Reply:
Also, isn't this set automatically when the image is built on a 64bit system?

Reply:
Depending on the locally used `e2fsprogs` release it might, but not necessarily (defaults have changed over time). For 64bit target architectures and reproducibility purposes I think it's best to explicityly enable it.

I've committed the suggested change.